src/mlfe_typer.erl

```diff
@@ -1010,8 +1010,10 @@ typ_module(#mlfe_module{functions=Fs,
                     FunRes = typ_module_funs(Fs, Env3, []),
 
                     case type_module_tests(Tests, Env3, ok, FunRes) of
-                        {error, _} = E -> E;
-                        [_|_] = Funs   -> {ok, M#mlfe_module{functions=Funs}}
+                        {error, _} = Err        ->
+                          Err;
```

Comment:
Minor nit-pick on indentation here - are you using an emacs or vim mode for Erlang?